usecases.bs

```diff
@@ -19,112 +22,200 @@ Inline Github Issues: true
+            super.onreading = () => {
+                this.inFreeFall_ = Math.hypot(super.x, super.y, super.z) < 0.5;
+                func();
+            }
+        }
+
+        get inFreeFall {
+            return this.inFreeFall_;
+        }
+    }
+    </pre>
+</div>
 
-Turn by turn navigation {#turn-navigation}
-------------------
+When the web application requires measurements that are fused from multiple sensor sources, 
+simple aggregation pattern may be used.
 
+<div class="example">
+    In this simplified example, we define FusionSensor class that aggregates multiple sources and
+    provides fused data.
+
+    <pre highlight="js">
+    class FusionSensor {
+        constructor(sensors, algorithm) {
+            this.sensors_ = sensors;
+            this.algorithm_ = algorithm.bind(this);
+        }
+
+        start() {
+            this.sensors_.map(sensor => sensor.start());
+        }
+
+        stop()  {
+            this.sensors_.map(sensor => sensor.stop());
+        }
+
+        set onreading(func) {
+            this.sensors_.map(sensor => {
+                sensor.onreading = () => {
+                    this.fusedReading_ = this.algorithm_(sensor, this.sensors_);
+                    func();
+                }
+            });
+        }
+
+        get fusedReading() {
+            return this.fusedReading_;
+        }
+    }
+
+    // Stateful algorithm that finds brightest spot during calibration phase and uses
+    // it as a fixed frame of reference to calculate relative orientation from brigtest
+    // light source.
+    let custom_algo = (changed_source, all_sources) => { return new DOMMatrix(); }
+    let source_sensors = [new Accelerometer(), new Gyroscope(), new AmbientLightSensor()];
+    let sensor = new FusionSensor(source_sensors, custom_algo);
+    sensor.onreading = () => { console.log(JSON.stringify(sensor.fusedReading)); }
+    sensor.start();
+    </pre>
+</div>
 
-Pedometer {#pedometer}
---------------
+## Trigger sensors ## {#trigger-sensors}
 
-Issue(12):
+Issue: This group of sensors may not need frequency. The HW will just fire interrupt to
+signal about the change. If trigger sensor functionality is provided by e.g., accelerometer,
+sensor would be operating in special low-power mode.
 
+For the purpose of this document, trigger sensors group is characterised by the simplicity
+of the hardware, limited sensor configuration capabilities, special mode of operation or
+notification mechanism for the new sensor readings. For example, reed switch or hall effect sensors
+that will trigger hardware interrupt and provide one bit reading data (on / off).
 
-Promimity sensors {#proximity}
--------------
+The following list of sensors might be a good example of trigger sensors:
 
+- Reed switch and hall effect sensors
+- Significant motion, no motion, tap, flat, free fall, low-g and high-g sensors
+- Laser or infrared switch sensors
 
 Security and Privacy Considerations {#security-and-privacy}
```

Comment:
We can remove this section from this use cases document, needed only for API specs.